pkgs/development/compilers/gcc/common/builder.nix

```diff
@@ -210,112 +209,136 @@ originalAttrs:
     '';
 
     preInstall =
+      # What follows is a horribly cursed hack.
+      #
+      # GCC will install its libraries to $out/lib, $out/lib32, $out/lib64,
+      # $out/$targetConfig/lib, $out/$targetConfig/lib32 or $out/$targetConfig/lib64,
+      # depending on whether it's built as native or cross, and the exact target spec.
+      #
+      # We can't predict what it's actually going to do, and we also can't just tell it
+      # to always install to $out/lib, but we want everything to end up in $out/lib
+      # for consistency (multilib weirdness aside).
+      #
+      # So, we create a bunch of symlinks before we run GCC's install phase,
+      # redirecting every possible directory it may want to write to to the place
+      # we actually want things to be installed.
+      # We will then nuke the symlinks in postInstall.
+      #
+      # FIXME: there must be a better way to do this.
       ''
-        mkdir -p "$out/''${targetConfig}/lib"
-        mkdir -p "''${!outputLib}/''${targetConfig}/lib"
+        declare -ga compatibilitySymlinks=()
+
+        makeCompatibilitySymlink() {
+          for output in "$out" "''${!outputLib}"; do
+            local srcPath="$1"
+            local dstPath="$output/$2"
```

Comment:
Imho the terms of `src` and `dst` are somewhat ill-suited here, as I'm reading it as "make (a) compatibility symlink (from) `src` to `dst`", but the link is pointing in the other direction. The manpage of `ln` uses TARGET and LINK_NAME instead, and I find that to be a lot less confusing. How about using these terms here as well?

Reply:
Went with `linkTarget` and `linkName`, which is slightly less consistent with `ln`, but "target" alone is still confusing IMO.